File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.cpp
class Solution {
public:
const long long MOD = 1e9;
    int chalkReplacer(vector<int>& chalk, int k) {
        
         //long su=accumulate(chalk.begin(),chalk.end(),0);
         long long su=0;
         for(auto x:chalk){
            su+=x;
         }
        //su=(su)%MOD;
         k=k%su;
         if(k==0 || k<chalk[0]) return 0;

         

        int n=chalk.size();

         for(int i=0;i<n;i++){
            if(k>=chalk[i]){
                k-=chalk[i];
            }else{
                return i;
            }
         }

        return 0;
    }
};

File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:

        su=sum(chalk)
        k=(int)(k%su)
        if(k==0 or k<=chalk[0]):
            return 0

        for i in range(len(chalk)):
            if(k>=chalk[i]):
                k-=chalk[i]
            else:
                return i

        return 0

        